packages/@aws-cdk/aws-rds/lib/parameter-group.ts

```diff
@@ -76,60 +73,88 @@ export interface ParameterGroupProps {
 }
 
 /**
- * A parameter group
+ * A parameter group.
+ * Represents both a cluster parameter group,
+ * and an instance parameter group.
  *
  * @resource AWS::RDS::DBParameterGroup
  */
-export class ParameterGroup extends ParameterGroupBase {
+export class ParameterGroup extends Resource implements IParameterGroup {
   /**
-   * The name of the parameter group
+   * Imports a parameter group
    */
-  public readonly parameterGroupName: string;
-
-  constructor(scope: Construct, id: string, props: ParameterGroupProps) {
-    super(scope, id);
+  public static fromParameterGroupName(scope: Construct, id: string, parameterGroupName: string): IParameterGroup {
+    class Import extends Resource implements IParameterGroup {
+      public readonly parameterGroupName = parameterGroupName;
 
-    this.parameters = props.parameters ? props.parameters : {};
+      public bindToCluster(_options: ParameterGroupClusterBindOptions): ParameterGroupClusterConfig {
+        return { parameterGroupName };
+      }
 
-    const resource = new CfnDBParameterGroup(this, 'Resource', {
-      description: props.description || `Parameter group for ${props.family}`,
-      family: props.family,
-      parameters: Lazy.anyValue({ produce: () => this.parameters }),
-    });
+      public bindToInstance(_options: ParameterGroupInstanceBindOptions): ParameterGroupInstanceConfig {
+        return { parameterGroupName };
+      }
+    }
 
-    this.parameterGroupName = resource.ref;
+    return new Import(scope, id);
   }
-}
 
-/**
- * Construction properties for a ClusterParameterGroup
- */
-// tslint:disable-next-line:no-empty-interface
-export interface ClusterParameterGroupProps extends ParameterGroupProps {
-
-}
-/**
- * A cluster parameter group
- *
- * @resource AWS::RDS::DBClusterParameterGroup
- */
-export class ClusterParameterGroup extends ParameterGroupBase {
   /**
-   * The name of the parameter group
+   * Parameters of the parameter group
    */
-  public readonly parameterGroupName: string;
+  protected parameters?: { [key: string]: string } = {};
+  private readonly family: string;
+  private readonly description?: string;
 
-  constructor(scope: Construct, id: string, props: ClusterParameterGroupProps) {
+  private clusterCfnGroup?: CfnDBClusterParameterGroup;
+  private instanceCfnGroup?: CfnDBParameterGroup;
+
+  constructor(scope: Construct, id: string, props: ParameterGroupProps) {
     super(scope, id);
 
+    this.family = props.family;
+    this.description = props.description;
     this.parameters = props.parameters ? props.parameters : {};
+  }
+
+  public bindToCluster(_options: ParameterGroupClusterBindOptions): ParameterGroupClusterConfig {
+    if (!this.clusterCfnGroup) {
+      const parentScope = this.instanceCfnGroup ?? this;
```

Comment:
This maybe deserves a comment. Why parent the scope of one param group onto the other?

Reply:
It's because I want both of the `Cfn*` resources to have the ID `'Resource'`, but if one of the `bind*()` methods was already called, that would result in a duplicate identifier in the `this` scope. Hence this trick.

Reply:
¯\_(ツ)_/¯ I'd add a comment to that effect, unless this is a super-common pattern I haven't seen before.

Reply:
I was also thinking - instead of changing the parent, I could change the ID.

So, this would become:

```ts
  public bindToCluster(_options: ParameterGroupClusterBindOptions): ParameterGroupClusterConfig {
    if (!this.clusterCfnGroup) {
      const resourceId = this.instanceCfnGroup ? 'SecondResource' : 'Resource';
      this.clusterCfnGroup = new CfnDBClusterParameterGroup(this, resourceId, {
        description: this.description || `Cluster parameter group for ${this.family}`,
        family: this.family,
        parameters: Lazy.anyValue({ produce: () => this.parameters }),
      });
```

Is that more clear @njlynch ?

Reply:
Any reason not to just do this?
```ts
public bindToCluster(...) {
  this.clusterCfnGroup = new CfnDBClusterParameterGroup(this, 'ClusterParamGroup', {
  ...
  });
}
public bindToInstance(...) {
  this.clusterCfnGroup = new CfnDBClusterParameterGroup(this, 'InstanceParamGroup', {
  ...
  });
}
```

Reply:
i find them both equally readable, but a comment will be handy for posterity as I think it won't be intuitive to a new contributor why we do this

Reply:
> Any reason not to just do this?
> 
> ```ts
> public bindToCluster(...) {
>   this.clusterCfnGroup = new CfnDBClusterParameterGroup(this, 'ClusterParamGroup', {
>   ...
>   });
> }
> public bindToInstance(...) {
>   this.clusterCfnGroup = new CfnDBClusterParameterGroup(this, 'InstanceParamGroup', {
>   ...
>   });
> }
> ```

Actually, yes.

Since the existing classes use the `Resource` IDs, changing them now to `'ClusterParamGroup'`/`'InstanceParamGroup'` would cause a replacement of them in customer's CloudFormation Stacks after they upgrade their version of CDK, which is quite a disruptive change.

Also, the name `'Resource'` (and `'Default'`) is treated a little bit especially - it's the return value from the `this.node.defaultChild` method (which is often used with [escape hatches](https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html)).